pkg/alertmanager/multitenant.go

```diff
@@ -435,11 +436,17 @@ func (am *MultitenantAlertmanager) setConfig(cfg alerts.AlertConfigDesc) error {
 		am.alertmanagersMtx.Unlock()
 	} else if am.cfgs[cfg.User].RawConfig != cfg.RawConfig || hasTemplateChanges {
 		level.Info(am.logger).Log("msg", "updating new per-tenant alertmanager", "user", cfg.User)
+		activeBefore := existing.IsActive()
 		// If the config changed, apply the new one.
 		err := existing.ApplyConfig(cfg.User, userAmConfig, rawCfg)
 		if err != nil {
 			return fmt.Errorf("unable to apply Alertmanager config for user %v: %v", cfg.User, err)
 		}
+
+		// If we went from paused to active, we need to re-add an user-regitry.
+		if !activeBefore && existing.IsActive() {
+			am.alertmanagerMetrics.addUserRegistry(cfg.User, prometheus.NewRegistry())
```

Comment:
I can't understand this change honestly. You're just setting a new prometheus registry to `alertmanagerMetrics` but that registry will never get alertmanager metrics registered on.

I still believe this PR has not been conceived correctly. What I think you should do is completely release the alertmanager resources once a tenant is removed from this replica, and create a new alertmanager (with a new registry which is then registered to `alertmanagerMetrics`) when the tenant is re-added back. Basically this should completely remove the `IsActive()` logic.

Reply:
> I can't understand this change honestly. You're just setting a new prometheus registry to alertmanagerMetrics but that registry will never get alertmanager metrics registered on.

Sorry Marco, I completely left this PR in an abandoned state but my ultimate intention was to close it. I quickly realised exactly what you pointed out that the new registry is never initialised with the corresponding metrics.

Given I'm doing work around the area with the scalability, and completely removing the `IsActive` is a bit more involved than I initially expect, I'll close this and defer to later.